public/scripts/dashboard.js

```diff
@@ -1,15 +1,7 @@
 function getDashboardData() {
-    db.collection(collectionName).onSnapshot(function(snapshot) {
-        snapshot.docChanges().forEach(function(change) {
-            if (change.type === "added") {
-                console.log(change.doc.data());
-                createDashboardInstance(change.doc.data());
-            }
-            if (change.type === "modified") {
-                console.log("modified");
-                // updateDashboard(change.doc.data());
-            }
-        });
+    db.collection(collectionName).get().then(function(doc) {
+        console.log(doc.data());
+        createDashboardInstance(doc.data());
```

Comment:
The logic can be updated here based on what is needed to be shown in db.

Reply:
And under the logical improvement no more dashboardDataset will be used and therefore making the implementation very simple. Please think and update properly.

Reply:
And at the end you must call getTask(), Please think it's simple design change and lot's of code will reduce.

Reply:
okay